Gateway/Request/Payment/AuthorizeRequest.php

```diff
@@ -80,8 +80,14 @@ public function build(array $buildSubject)
             }
         }
 
+        if ($this->payment->hasAdditionalInformation(PaymentDataAssignObserver::BLIK_CODE)
+            && ! empty($this->payment->getAdditionalInformation(PaymentDataAssignObserver::BLIK_CODE))) {
+            $request['body'][PaymentField::AUTHORIZATION_CODE] = $this->payment
+                ->getAdditionalInformation(PaymentDataAssignObserver::BLIK_CODE);
+        }
+
         $request['headers'] = [
-            PaymentField::IDEMPOTENCY_KEY_FIELD_NAME => uniqid($referenceId, true)
+            PaymentField::IDEMPOTENCY_KEY_FIELD_NAME => uniqid(substr($referenceId, 0, 22), true)
```

Comment:
dozwolone będzie 45 znaków

Reply:
uniqid z opcją more_entropy = true zwraca 23 znaki , a substr($referenceId, 0, 22) to prefix

Reply:
nie wiem czy to poprawiać, moim zdaniem wynik tego co jest bedzie miał maks 45 znaków i dużą unikalność